rsocket-core/src/main/java/io/rsocket/frame/decoder/DefaultPayloadDecoder.java

```diff
@@ -11,7 +11,7 @@
 class DefaultPayloadDecoder implements PayloadDecoder {
 
   @Override
-  public Payload apply(ByteBuf byteBuf) {
+  public synchronized Payload apply(ByteBuf byteBuf) {
```

Comment:
Hi, @robertroeser! I've run a benchmark with multiple clients sending messages concurrently, and lock profiler (a part of [async-profiler](https://github.com/jvm-profiling-tools/async-profiler)) showed that there was lock contention in `DefaultPayloadDecoder` which seems to be introduced in this PR. 

<img width="920" alt="Screen Shot 2019-11-29 at 6 42 24 pm" src="https://user-images.githubusercontent.com/1780970/69852293-038c8c80-12d8-11ea-8001-196309724bb7.png">

I've looked through the code, but it's not obvious to me why it requires synchronisation since the class seems to be stateless. So, I'm wondering whether this `synchronized` can be safely removed in which case I'll be happy to send a PR or it serves some reason. 

Thanks :)

Reply:
Hello, @SerCeMan!

Thanks for finding that. I guess that was done by mistake and there is no need to have `synchronized` keyword here.